src/server/session.ts

```diff
@@ -2914,7 +2914,7 @@ namespace ts.server {
             return { response, responseRequired: true };
         }
 
-        private handlers = new Map(getEntries<(request: protocol.Request) => HandlerResponse>({
+        private handlers = new Map(getEntries<(request: any) => HandlerResponse>({ // TODO(jakebailey): correctly type the handlers

```

Comment:
I tried to do something more clever here, but was foiled by how this file declares `CommandNames`.

Reply:
I sure do hate `CommandNames`! If you change this to take `unknown`, does it break anything?

Reply:
Yes, because functions like `(request: protocol.UpdateOpenRequest) => ...` aren't assignable to `(request: unknown) => ...`.


```
Argument of type '{ [x: number]: ((request: protocol.OpenExternalProjectRequest) => HandlerResponse) | ((request: protocol.OpenExternalProjectsRequest) => HandlerResponse) | ((request: protocol.CloseExternalProjectRequest) => HandlerResponse) | ((request: protocol.SynchronizeProjectListRequest) => HandlerResponse) | ... 53 more ... |...' is not assignable to parameter of type 'MapLike<(request: unknown) => HandlerResponse>'.
  'number' and 'string' index signatures are incompatible.
    Type '((request: OpenExternalProjectRequest) => HandlerResponse) | ((request: OpenExternalProjectsRequest) => HandlerResponse) | ... 55 more ... | ((request: InlayHintsRequest) => HandlerResponse)' is not assignable to type '(request: unknown) => HandlerResponse'.
      Type '(request: protocol.OpenExternalProjectRequest) => HandlerResponse' is not assignable to type '(request: unknown) => HandlerResponse'.
        Types of parameters 'request' and 'request' are incompatible.
          Type 'unknown' is not assignable to type 'OpenExternalProjectRequest'.ts(2345)
```

What I wanted to do was something like https://github.com/microsoft/vscode-python/blob/912ad3226cf51b06f77a44adb30f8614fbfe5ac3/src/client/telemetry/index.ts#L311 where we define what this handler map should be based on an interface that can map commands to their request/responses.

Reply:
I have a change that improves this for the request types, but the way that the response types work is very weird; the handlers return an object that looks like `{ response: ... }`, then that response is then transformed and used as `body` later, so using the `*Response` types won't quite work.

Potentially if I instead used the `*Body` types to define what the return types should be.

Reply:
I have a better fix stashed for this, however, it exposes the fact that some of our handlers are non-compliant, and I don't think it's a good idea to fix all of those handlers in this PR just to get rid of this `any`.